File: src/main/java/play/pluv/playlist/infra/dto/OcrMusicRequest.java
package play.pluv.playlist.infra.dto;

import java.util.List;

public record OcrMusicRequest(
    List<Base64EncodedImage> images
) {

  public static OcrMusicRequest from(final List<String> base64EncodedImages) {
    return new OcrMusicRequest(
        base64EncodedImages.stream()
            .map(Base64EncodedImage::new)
            .toList()
    );
  }

  private record Base64EncodedImage(String base64EncodedImage) {

  }
}

File: src/main/kotlin/play/pluv/playlist/infra/dto/OcrMusicRequest.kt
package play.pluv.playlist.infra.dto

data class OcrMusicRequest(
    val images: List<Base64EncodedImage>
) {

    data class Base64EncodedImage(val base64EncodedImage: String)

    companion object {
        fun from(base64EncodedImages: List<String>): OcrMusicRequest {
            return OcrMusicRequest(
                images = base64EncodedImages.map { Base64EncodedImage(it) }
            )
        }
    }
}
